micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/LicenseKeyConfiguration.java

```diff
@@ -0,0 +1,99 @@
+package info.novatec.micronaut.camunda.bpm.feature;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.event.ApplicationEventListener;
+import io.micronaut.runtime.server.event.ServerStartupEvent;
+import org.camunda.bpm.engine.ProcessEngine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Optional;
+import java.util.Scanner;
+
+/**
+ * @author Martin Sawilla
+ */
+// Implementation based on https://github.com/camunda/camunda-bpm-spring-boot-starter/blob/master/starter/src/main/java/org/camunda/bpm/spring/boot/starter/configuration/impl/custom/EnterLicenseKeyConfiguration.java
+@Singleton
+@Requires(property = "camunda.bpm.license-file")
+public class LicenseKeyConfiguration implements ApplicationEventListener<ServerStartupEvent> {
+    private static final Logger log = LoggerFactory.getLogger(LicenseKeyConfiguration.class);
+    protected static String DEFAULT_LICENSE_FILE = "camunda-license.txt";
+    protected static String HEADER_FOOTER_REGEX = "(?i)[-\\s]*(BEGIN|END)\\s*(OPTIMIZE|CAMUNDA|CAMUNDA\\s*BPM)\\s*LICENSE\\s*KEY[-\\s]*";
+
+    protected Configuration configuration;
+
+    protected ProcessEngine processEngine;
+
+    protected CamundaBpmVersion camundaBpmVersion;
+
+    public LicenseKeyConfiguration(Configuration configuration, ProcessEngine processEngine, CamundaBpmVersion camundaBpmVersion) {
+        this.configuration = configuration;
+        this.processEngine = processEngine;
+        this.camundaBpmVersion = camundaBpmVersion;
+    }
+
+    @Override
+    public void onApplicationEvent(ServerStartupEvent event) {
+
+        if(!camundaBpmVersion.isEnterprise()) {
+            log.info("You are not using the EE Version. No need to provide a license.");
+            return;
+        }
+
+        Optional<String> licenseKey = Optional.empty();
+
+        if(configuration.getLicenseFile().isPresent()) {
```

Comment:
I see that you copied the logic but I think it could be more precise and also throw errors if they occur. Let's discuss.